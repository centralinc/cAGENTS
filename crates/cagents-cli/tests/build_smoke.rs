use assert_cmd::Command;
use assert_fs::prelude::*;
use predicates::prelude::*;
use std::fs;

#[test]
fn test_build_creates_agents_md() {
    // Create a temporary copy of examples/basic
    let temp = assert_fs::TempDir::new().unwrap();
    let workspace_root = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))
        .parent()
        .unwrap()
        .parent()
        .unwrap()
        .to_path_buf();
    let basic_src = workspace_root.join("examples/basic");

    // Copy cAGENTS directory
    let cagents_src = basic_src.join(".cAGENTS");
    let cagents_dst = temp.path().join(".cAGENTS");
    copy_dir_recursive(&cagents_src, &cagents_dst).expect("Failed to copy cAGENTS");

    // Copy source files so glob-based templates have files to match
    let src_dir = basic_src.join("src");
    if src_dir.exists() {
        let dst_src = temp.path().join("src");
        fs::create_dir_all(&dst_src).unwrap();
        for entry in fs::read_dir(&src_dir).unwrap() {
            let entry = entry.unwrap();
            if entry.file_type().unwrap().is_file() {
                fs::copy(entry.path(), dst_src.join(entry.file_name())).unwrap();
            }
        }
    }

    let tests_dir = basic_src.join("tests");
    if tests_dir.exists() {
        let dst_tests = temp.path().join("tests");
        fs::create_dir_all(&dst_tests).unwrap();
        for entry in fs::read_dir(&tests_dir).unwrap() {
            let entry = entry.unwrap();
            if entry.file_type().unwrap().is_file() {
                fs::copy(entry.path(), dst_tests.join(entry.file_name())).unwrap();
            }
        }
    }

    // Run cagents build
    let mut cmd = Command::cargo_bin("cagents").unwrap();
    cmd.current_dir(temp.path())
        .arg("build")
        .assert()
        .success();

    // Check that AGENTS.md was created
    let agents_md = temp.child("AGENTS.md");
    agents_md.assert(predicate::path::exists());

    // Read and verify content
    let content = fs::read_to_string(agents_md.path()).unwrap();

    // Should contain content from project-meta template
    assert!(content.contains("Project Context"), "Should contain Project Context section");
    assert!(content.contains("Communication Style"), "Should contain template content");

    // Snapshot test for exact format
    insta::assert_snapshot!(content);
}

// Helper to recursively copy directories
fn copy_dir_recursive(src: &std::path::Path, dst: &std::path::Path) -> std::io::Result<()> {
    fs::create_dir_all(dst)?;
    for entry in fs::read_dir(src)? {
        let entry = entry?;
        let ty = entry.file_type()?;
        let src_path = entry.path();
        let dst_path = dst.join(entry.file_name());

        if ty.is_dir() {
            copy_dir_recursive(&src_path, &dst_path)?;
        } else {
            fs::copy(&src_path, &dst_path)?;
        }
    }
    Ok(())
}
