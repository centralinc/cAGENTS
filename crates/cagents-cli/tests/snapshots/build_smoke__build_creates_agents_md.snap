---
source: crates/cagents-cli/tests/build_smoke.rs
assertion_line: 67
expression: content
---
<!--
This file is auto-generated by cAGENTS. Do not edit directly.

To update: run `cagents build`
For file-specific context: run `cagents context <filename>`
-->

# Project Context

**Owner:** LocalOverride
**Communication Style:** concise
**Current Branch:** 

**Local Note:** This variable only exists in local config


When working on this project, maintain a concise communication style.

## TypeScript

- Prefer explicit types; avoid `any`.
- Run typecheck: `pnpm typecheck`.
- See [System Models](docs/system-design/models.md) for domain entities.

## Rust Code Guidelines

**For this directory:**

- Use `rustfmt` before committing
- Run `cargo clippy` and fix all warnings
- Add doc comments (`///`) for public items
- Use `#[must_use]` for functions that return important values

---

## Working with cAGENTS

This file was generated by cAGENTS. To work efficiently with this codebase:

1. **Run once at start of session**: `cagents build` ensures all agent rules are up-to-date
2. **Get file-specific context**: `cagents context <filename>` retrieves all relevant rules for a specific file - more efficient than reading this entire file
3. **Best practice**: Re-run `cagents build` when context seems stale or after pulling changes

Learn more: https://github.com/centralinc/cagents
